### PR TITLE
prevent duplicate assembly steps

### LINKED ISSUE #27: Detect duplicate step names
Awesome library! One improvement came to mind. Since we're building steps as an array of objects, it would be nice for the library to detect if the user accidentally named two different steps with the same name, and raise an error. I just think that the way assemblies are built, this mistake can happen easily.

The current behaviour is probably that the latter step will silently override the previous, and I was just thinking that it would be great to explicitly let the user know that they made a mistake.

### LINKED ISSUE #49: Assembly List fails with Signature Error
Using a working Transloadit Ruby-SDK setup (processes video files correctly), I cannot seem to get the assembly.list or Transloadit.bill calls to work. I have "Deny requests that do not include a signature or a wrong one?" turned off on my account for now.

```
irb(main):001:0> assembly = Transloadit.assembly
=> {:auth=>{:key=>"MY_KEY_HERE", :expires=>"2018/05/19 01:24:57+00:00"}}
```

```
irb(main):002:0> assembly.list
=> {"error"=>"INVALID_SIGNATURE", "message"=>"The given signature does not match ours."}
```

I'm pretty sure it's because the signature is not being generated/sent but that's not made clear that it needs to be sent anywhere other than the transloadit docs.

Also, there's no documentation here on how to send a signature or if it's automatic.

Edit: Inspecting the request shows that it is indeed generating a signature, unclear why it's wrong.

### PR DESCRIPTION
**Pull request by ifedapoolarewaju**
fixes #27

### CODE REVIEW

**Comment by rmehner** [comment]
👍 This will be a breaking change tho, so we should have a CHANGELOG entry for this :)

**Comment by kvz** [comment]
+1 on CHANGELOG update, and I wonder if we want to bump the version to next major then also?

**Comment by rmehner** [comment]
> bump the version to next major then also

Yup, it's a breaking change :)

**Comment by ifedapoolarewaju** [comment]
@rmehner @kvz updated, thank you

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### 2.1.0 / ????-??-?? ###
```
Comment (rmehner) [inline CHANGELOG.md]: 3.0.0, since this is a breaking change.

**Comment by ifedapoolarewaju** [comment]
@rmehner done, thank you

**Comment by rmehner** [comment]
Thank you @ifedapoolarewaju. I can push out the release later today :)

**Comment by ifedapoolarewaju** [comment]
@rmehner to merge or not to merge?

**Comment by rmehner** [comment]
Totally forgot about this. Will put this into my calendar to push out the release later today.

**Pull request merged**